File: src/main/java/de/schramm/royalbash/controller/service/gameevent/NoOpEvent.java
package de.schramm.royalbash.controller.service.gameevent;

import de.schramm.royalbash.controller.service.core.Game;
import lombok.Builder;
import lombok.Value;

@Value
@Builder
public class NoOpEvent implements GameEvent {

    @Override
    public Game invoke(Game game) {
        return game;
    }
}

File: src/main/kotlin/de/schramm/royalbash/controller/service/gameevent/NoOpEvent.kt
package de.schramm.royalbash.controller.service.gameevent

import de.schramm.royalbash.controller.service.core.Game

data class NoOpEvent(val noopValue: String) : GameEvent {

    constructor(): this(noopValue = "")

    override fun invoke(game: Game): Game {
        return game
    }
}
